src/components/layout/TransactionWorkflow/TransactionWorkflow.tsx

```diff
@@ -0,0 +1,406 @@
+  ContractType,
+  CreateAntPayload,
+  REGISTRY_INTERACTION_TYPES,
+  RegistryInteraction,
+  TRANSACTION_DATA_KEYS,
+  TransactionData,
+} from '../../../types';
+import { isObjectOfTransactionPayloadType } from '../../../utils/searchUtils';
+import { AntCard } from '../../cards';
+import DeployTransaction from '../DeployTransaction/DeployTransaction';
+import TransactionComplete from '../TransactionComplete/TransactionComplete';
+import Workflow, { WorkflowStage } from '../Workflow/Workflow';
+
+export enum TRANSACTION_WORKFLOW_STATUS {
+  PENDING = 'pending',
+  CONFIRMED = 'confirmed',
+  SUCCESSFUL = 'successful',
+  FAILED = 'failed',
+}
+
+function TransactionWorkflow({
+  contractType,
+  interactionType,
+  transactionData,
+  workflowStage,
+}: {
+  contractType: ContractType;
+  interactionType: AntInteraction | RegistryInteraction;
+  transactionData: TransactionData;
+  workflowStage: TRANSACTION_WORKFLOW_STATUS;
+}) {
+  const [{ deployedTransactionId }, dispatchTransactionState] =
+    useTransactionState();
+  const [{ arweaveDataProvider }] = useGlobalState();
+  const { assetId, functionName, ...payload } = transactionData;
+  const navigate = useNavigate();
+  const [steps, setSteps] = useState<
+    | {
+        [x: number]: { title: string; status: string };
+      }
+    | undefined
+  >(() =>
+    getStepsByTransactionType({
+      contractType,
+      interactionType,
+    }),
+  );
+  const [stages, setStages] = useState<
+    { [x: string]: WorkflowStage } | undefined
+  >(() =>
+    getStagesByTransactionType({
+      contractType,
+      interactionType,
+    }),
+  );
+
+  useEffect(() => {
+    const newStages = getStagesByTransactionType({
+      contractType,
+      interactionType,
+    });
+    if (newStages) setStages(newStages);
+    if (
+      workflowStage === TRANSACTION_WORKFLOW_STATUS.CONFIRMED &&
+      deployedTransactionId
+    ) {
+      dispatchTransactionState({
+        type: 'setWorkflowStage',
+        payload: TRANSACTION_WORKFLOW_STATUS.SUCCESSFUL,
+      });
+    }
+  }, [workflowStage, deployedTransactionId, transactionData]);
+
+  async function handleStage(direction: string) {
+    try {
+      switch (direction) {
+        case 'next':
+          switch (workflowStage) {
+            case TRANSACTION_WORKFLOW_STATUS.PENDING: {
+              const newSteps = getStepsByTransactionType({
+                contractType,
+                interactionType,
+              });
+              newSteps['1'].status = 'success';
+              newSteps['2'].status = 'pending';
+              setSteps(newSteps);
+              dispatchTransactionState({
+                type: 'setWorkflowStage',
+                payload: TRANSACTION_WORKFLOW_STATUS.CONFIRMED,
+              });
+              const originalTxId = await arweaveDataProvider.writeTransaction(
+                new ArweaveTransactionID(assetId),
+                {
+                  function: functionName,
+                  payload,
```

Comment:
https://viewblock.io/arweave/tx/wcAQhzSnJ3OoA6QzLt9AihFx2p2uH5luiWDGC8mdFII

the payload needs to be spread

```suggestion
                  ...payload,
```